syntax = "proto3";

// MemberActionResponse is the message to state whether the action taken was successful or not
message MemberActionResponse {
    // success is a flag to indicate if the message was successfully sent.
    bool success = 1;
    // http status is the integer response of the request results whilst following the http code format
    int32 http_status = 2;
    // error is the error message if there was an error sending the message, it is empty if the message was successfully sent.
    repeated string error_message = 3;
}

// UserRequest contains the data for a user to take action on another user within a specific community
message UserRequest {
    // community_id is where the action is taking place
    int32 community_id = 1;
    // user_id is the user performing the action
    int32 user_id = 2;
    // action_user_id is the specified user to perform the action on
    int32 action_user_id = 3;
}

service MemberManagement {
    // PromoteUser is used for both promotion and demotion of a specific user
    rpc PromoteUser(UserRequest) returns (MemberActionResponse) {}
    // BanUser is used by admins to either ban or unban users
    rpc BanUser(UserRequest) returns (MemberActionResponse) {}
}
